install/install.js

```diff
@@ -112,13 +114,26 @@ function checkPage()
                 .error('<i class="fas fa-ban mr-2"></i>Administrator passwords are not similar.', 10)
                 .dismissOthers();
             return false;
-        } else{
-            $("#hid_db_pre").val($("#tbl_prefix").val());
-            jsonValues = {"tbl_prefix":sanitizeString($("#tbl_prefix").val()), "sk_path":sanitizeString($("#sk_path").val()), "admin_pwd":sanitizeString($("#admin_pwd").val()), "send_stats":""};
-            dataToUse = JSON.stringify(jsonValues);
-            tasks = ["misc*preparation"];
-            multiple = "";
         }
+        
+        // Email checks
+        if ($("#admin_email").val() === "") {
+            alertify
+                .error('<i class="fas fa-ban mr-2"></i>You must define an email for Administrator account.', 10)
+                .dismissOthers();
+            return false;
+        } else if (!/^[^\s@]+@[^\s@]+\.[^\s@]+$/.test($("#admin_email").val())) {
```

Comment:
This is not directly related to this change but it highlights it: The installation script seems to be vulnerable to SQL injection. The risk is mitigated since it is limited to the installation and we delete the file at the end.

Edit the request on the fly:
![image](https://github.com/user-attachments/assets/5deea1d5-b9fd-4f87-83d3-c76f65e9491b)
![image](https://github.com/user-attachments/assets/bf826c96-7071-4a9c-b052-1d85d7a0d352)

No php error but the table is empty:
![Capture d'écran 2024-10-01 092453](https://github.com/user-attachments/assets/b43d990c-6824-4e18-8939-599f9398176c)

I think other fields like saltkey might be vulnerables but need tests to validate.

Can you add email filter on the php side too?

Reply:
I had also made these observations, but I had the idea of ​​doing the mapping first and suggesting the necessary corrections in another PR.

@nilsteampassnet, What do you think?

Reply:
@evertton 
For my point of view the PR should be complete.
But on the other hand, the vulnerability was already there.
So I will merge the PR and proovide a fix in order to fix the vulnerability in global